src/core/Core.js

```diff
@@ -411,6 +424,10 @@ class Uppy {
     const removedFile = updatedFiles[fileID]
     delete updatedFiles[fileID]
 
+    this.removers.forEach((remover) => {
+      remover([fileID])
+    })
```

Comment:
Processor and uploader functions are kept in arrays in core because you can have multiple. I think a single remover function should be enough; it doesn't need a pipeline like uploads may need.

Maybe this can be done by emitting a `remove-file` event here in `removeFile()`, and then listening for it in a `DeleteFiles` plugin instead?